Sources/MediaNavigationBar.swift

```diff
@@ -0,0 +1,134 @@
+/*****************************************************************************
+ * MediaNavigationBar.swift
+ *
+ * Copyright © 2019 VLC authors and VideoLAN
+ *
+ * Authors: Robert Gordon <robwaynegordon # gmail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import AVKit
+
+@objc (VLCMediaNavigationBarDelegate)
+protocol MediaNavigationBarDelegate {
+    func didMinimizePlayback(_ navigationBar: MediaNavigationBar)
+    func didToggleChromeCast(_ navigationBar: MediaNavigationBar)
+    func didToggleAirPlay(_ navigationBar: MediaNavigationBar)
+}
+
+@objc (VLCMediaNavigationBar)
+@objcMembers class MediaNavigationBar: UIView {
+    
+    // MARK: Instance Variables
+    weak var delegate: MediaNavigationBarDelegate?
+    
+    lazy var minimizePlaybackButton: UIButton = {
+        var minButton = UIButton(type: .system)
+        minButton.addTarget(self, action: #selector(minimizePlayback), for: .touchUpInside)
+        minButton.setImage(UIImage(named: "iconChevron"), for: .normal)
+        minButton.tintColor = .white
+        minButton.translatesAutoresizingMaskIntoConstraints = false
+        return minButton
+    }()
+    
+    private var mediaTitleTextLabel: UILabel = {
+        var label = UILabel()
+        label.numberOfLines = 1
+        label.textColor = .white
+        label.font = UIFont(name: "SFProDisplay-Medium", size: 17)
+        label.text = NSLocalizedString("TITLE", comment: "Video Title")
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    
+    lazy var chromeCastButton: UIButton = {
+        var chromeButton = UIButton(type: .system)
+        chromeButton.addTarget(self, action: #selector(toggleChromeCast), for: .touchUpInside)
+        chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
+        chromeButton.tintColor = .white
+        chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        return chromeButton
+    }()
+    
+    @available(iOS 11.0, *)
+    lazy var airPlayRoutePickerViewButton: AVRoutePickerView = {
+        var airPlayRoutePicker = AVRoutePickerView(frame: .zero)
+        airPlayRoutePicker.activeTintColor = .orange
+        airPlayRoutePicker.tintColor = .white
+        airPlayRoutePicker.translatesAutoresizingMaskIntoConstraints = false
+        return airPlayRoutePicker
+    }()
```

Comment:
yeah just return different buttons here depending on the os version

Reply:
Should I return a regular UIButton?